DOC-12443: Reorganize GSI documents (#3672)
couchbase/docs-server: modules/learn/pages/services-and-indexes/indexes/indexes.adoc

The _Search_ and _Analytics_ services both provide their own indexes, internally.

Indexes, when well-designed, provide significant enhancements to the performance of search-operations.

== Indexes

The following forms of index are available:

Primary:: Provided by the xref:services-and-indexes/services/index-service.adoc[Index Service], this is based on the unique key of every item in a specified collection.
Every primary index is maintained asynchronously.
A primary index is intended to be used for simple queries, which have no filters or predicates.
For information on primary indexes, see xref:indexes:indexing-overview.adoc[].

Secondary:: Provided by the xref:services-and-indexes/services/index-service.adoc[Index Service], this is based on an attribute _within_ a document.
The value associated with the attribute can be of any type: scalar, object, or array.
+
A Secondary Index is frequently referred to as a _Global Secondary Index_, or _GSI_.
This is the kind of index used most frequently in Couchbase Server, for queries performed with {sqlpp}.
For information on Global Secondary Indexes, see xref:indexes:indexing-overview.adoc[].

> Search:: Provided by the xref:services-and-indexes/services/search-service.adoc[Search Service], this is a specially purposed index, which contains targets derived from the textual contents of documents within one or more specified keyspaces.

Reviewer: ```suggestion
Search:: Provided by the xref:services-and-indexes/services/search-service.adoc[Search Service], this is a specially purposed index, which contains targets derived from the contents of documents within one or more specified keyspaces.
```

> Text-matches of different degrees of exactitude can be searched for.

Reviewer: ```suggestion
Search indexes support text matching, geospatial, date-time, numeric range searches, and more.
```

> Both input and target text-values can be purged of irrelevant characters (such as punctuation marks or HTML tags).

Reviewer: ```suggestion
For text matching, you can add filters to remove undesirable characters from input and target text values, such as punctuation marks or HTML tags.
```

> For information on how to create Search indexes, see xref:fts:fts-creating-indexes.adoc[Creating Indexes].

Reviewer: ```suggestion
For information on how to create Search indexes, see xref:search:create-search-indexes.adoc[].
```
